### PR TITLE
Google calendar synchronisation

### PR DESCRIPTION
**Pull request by Mathieu414**
Steps to setup the connection : 
- create a service account and download the keys in JSON format : https://naich.net/wordpress/index.php/using-the-google-calendar-api-from-your-web-site-with-php/. Next, allow permission for the service account to the targetted calendar.
- load you keys file to the root folder of the project
- create .env variables : GOOGLE_CREDENTIAL_PATH and GOOGLE_CALENDAR_ID

Things that could be improved : 
- better formatting of description. Going from markdown to html format.
- have several calendar for event types, especially simple events (course/entraînement).

### CODE REVIEW

File: app/services/GoogleCalendarService.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
+<?php
+
+use Google\Client;
+use Google\Service\Calendar as GoogleCalendarService;
+use Google\Service\Calendar\Event as GoogleCalendarEvent;
+class GoogleCalendar
```
Comment (Atmos4) [inline app/services/GoogleCalendarService.php:6]: Would be nice to name this GoogleCalendarService for clarity

File: database/migrations_sqlite/Version20241228180639.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public function down(Schema $schema): void
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->addSql('CREATE TEMPORARY TABLE __temp__orm_events AS SELECT id, name, start_date, end_date, deadline, description, open, bulletin_url, type FROM orm_events');
+        $this->addSql('DROP TABLE orm_events');
+        $this->addSql('CREATE TABLE orm_events (id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, name VARCHAR(255) NOT NULL, start_date DATETIME NOT NULL, end_date DATETIME NOT NULL, deadline DATETIME NOT NULL, description CLOB NOT NULL, open BOOLEAN NOT NULL, bulletin_url VARCHAR(255) NOT NULL, type VARCHAR(255) DEFAULT \'COMPLEX\' NOT NULL)');
+        $this->addSql('INSERT INTO orm_events (id, name, start_date, end_date, deadline, description, open, bulletin_url, type) SELECT id, name, start_date, end_date, deadline, description, open, bulletin_url, type FROM __temp__orm_events');
+        $this->addSql('DROP TABLE __temp__orm_events');
```
Comment (Atmos4) [inline database/migrations_sqlite/Version20241228180639.php:34]: It always scares me quite a lot to see how many hoops you need to jump in order to alter table in SQLite :D

File: composer.json [hunk 3]
```diff
@@ -38,10 +39,16 @@
     "seed": "php bin/seed",
     "db:migrate": "@php bin/doctrine --no-interaction migrations:migrate",
     "db:diff": "@php bin/doctrine --no-interaction migrations:diff",
-    "test": "@php bin/phpunit"
+    "test": "@php bin/phpunit",
+    "pre-autoload-dump": "Google\\Task\\Composer::cleanup"
```
Comment (Atmos4) [inline composer.json:43]: not sure what this does?

**Review by Atmos4 — COMMENTED** [review]
Really good job with this new service, I like the structure of the code!

**Review by Atmos4 — APPROVED** [review]
Imma approve now since all comments are really minor

File: database/migrations_sqlite/Version20241228180639.php (hunk 2 above)
Comment (Mathieu414) [reply database/migrations_sqlite/Version20241228180639.php:34]: Yep curious that you cannot alter the schema directly

File: composer.json (hunk 3 above)
Comment (Mathieu414) [reply composer.json:43]: This command is removing all the google libraries concerning the other APIs than those we need : Calendar in this case.

File: app/services/GoogleCalendarService.php (hunk 1 above)
Comment (Mathieu414) [reply app/services/GoogleCalendarService.php]: Right, just did that

**Pull request merged**